tests/0045-subscribe_update.c

```diff
@@ -821,6 +821,107 @@ static void do_test_resubscribe_with_regex() {
         SUB_TEST_PASS();
 }
 
+/**
+ * @brief Create many topics and apply several subscription
+ *        updates, unsubscribing and re-subscribing too.
+ *        After changing some subscriptions verifies that the assignment
+ *        corresponds to last one.
+ *
+ * @param with_rebalance_cb Use a rebalance callback to perform the assignment.
+ *                          It needs to poll the consumer when awaiting for the
+ *                          assignment in this case.
+ */
+static void do_test_subscribe_many_updates(rd_bool_t with_rebalance_cb) {
+#define TOPIC_CNT 100
+        char *topics[TOPIC_CNT] = {0};
+        char *topic;
+        size_t i;
+        char *group;
+        rd_kafka_t *rk;
+        rd_kafka_conf_t *conf;
+        const int max_subscription_size = 5, partition_cnt = 4;
+        rd_kafka_topic_partition_list_t *expected_assignment = NULL;
+
+        SUB_TEST("%s", with_rebalance_cb ? "with rebalance callback"
+                                         : "without rebalance callback");
+
+        RD_ARRAY_FOREACH(topic, topics, i) {
+                topics[i] = rd_strdup(test_mk_topic_name("topic", 1));
+        }
+        group = topics[0];
+
+        test_conf_init(&conf, NULL, 60);
+        if (with_rebalance_cb)
+                rd_kafka_conf_set_rebalance_cb(conf, test_rebalance_cb);
+        rk = test_create_consumer(group, NULL, conf, NULL);
+
+        TEST_SAY("Creating %d topics\n", TOPIC_CNT);
+        TEST_CALL_ERR__(test_CreateTopics_simple(rk, NULL, topics, TOPIC_CNT,
+                                                 partition_cnt, NULL));
+        test_wait_topic_exists(rk, topics[TOPIC_CNT - 1], 5000);
+        /* Give the cluster some more time to propagate metadata
+         * for TOPICS_CNT topics */
+        rd_sleep(1);
+
+        RD_ARRAY_FOREACH(topic, topics, i) {
+                size_t j;
+                int k,
+                    subscription_size =
+                        RD_MIN(max_subscription_size, TOPIC_CNT - i),
+                    expected_assignment_cnt = subscription_size * partition_cnt;
+
+                rd_kafka_topic_partition_list_t *subscription =
+                    rd_kafka_topic_partition_list_new(subscription_size);
+                rd_bool_t check_expected_assignment =
+                              (i % 5 == 0 || i == TOPIC_CNT - 1),
+                          do_unsubscribe = i % 7 == 0;
```

Comment:
Its more readable if we create each variable in a separate line.
```suggestion
                size_t j;
                int k;
                int subscription_size =
                        RD_MIN(max_subscription_size, TOPIC_CNT - i);
                int expected_assignment_cnt = subscription_size * partition_cnt;
                rd_kafka_topic_partition_list_t *subscription =
                    rd_kafka_topic_partition_list_new(subscription_size);
                rd_bool_t check_expected_assignment =
                              (i % 5 == 0 || i == TOPIC_CNT - 1);
                rd_bool_t do_unsubscribe = i % 7 == 0;
```